qutebrowser/mainwindow/tabwidget.py

```diff
@@ -733,9 +730,17 @@ def _get_icon_rect(self, opt, text_rect):
                      else QIcon.Disabled)
         icon_state = (QIcon.On if opt.state & QStyle.State_Selected
                       else QIcon.Off)
-        tab_icon_size = opt.icon.actualSize(icon_size, icon_mode, icon_state)
-        tab_icon_size = QSize(min(tab_icon_size.width(), icon_size.width()),
-                              min(tab_icon_size.height(), icon_size.height()))
+        # reserve space for favicon when tab bar is vertical (issue #1968)
+        position = config.get('tabs', 'position')
+        show_favicons = config.get('tabs', 'show-favicons')
+        if (opt.icon.isNull()
+            and position in [QTabWidget.East, QTabWidget.West]
+            and show_favicons):
```

Comment:
Some small style changes here:

- Please move the `and` to the end of the previous lines (the [PEP8 recommandation](https://www.python.org/dev/peps/pep-0008/#should-a-line-break-before-or-after-a-binary-operator) about this changed somewhat recently, but qutebrowser still uses the old style, so it's better to be consistent)
- Please indent the lines belonging to the `if` by another four spaces, so they're clearly separated from the if-body below
- You can simply use `config.get('tabs', 'show-favicons')` inline here, as the line wouldn't get too long.